src/vector/jitsi/index.ts

```diff
@@ -69,28 +119,84 @@ let widgetApi: WidgetApi;
         displayName = qsParam('displayName', true);
         avatarUrl = qsParam('avatarUrl', true); // http not mxc
         userId = qsParam('userId');
+        jitsiAuth = qsParam('auth', true);
+        roomId = qsParam('roomId', true);
 
         if (widgetApi) {
             await widgetApi.waitReady();
             await widgetApi.setAlwaysOnScreen(false); // start off as detachable from the screen
-        }
 
-        // TODO: register widgetApi listeners for PTT controls (https://github.com/vector-im/riot-web/issues/12795)
-
-        document.getElementById("joinButton").onclick = () => joinConference();
+            // See https://github.com/matrix-org/prosody-mod-auth-matrix-user-verification
+            if (jitsiAuth === 'openidtoken-jwt') {
+                window.addEventListener('message', onWidgetMessage);
```

Comment:
we shouldn't be doing this here - we should ensure that the `WidgetApi` class, for as long as it will survive, gets the support it needs to do these things.